src/panels/config/integrations/integration-panels/zwave_js/zwave_js-config-router.ts

```diff
@@ -33,14 +32,40 @@ class ZWaveJSConfigRouter extends HassRouterPage {
 
   @property({ type: Boolean }) public narrow = false;
 
-  private _configEntry = new URLSearchParams(window.location.search).get(
-    "config_entry"
-  );
+  private _configEntry: string | null = null;
 
   protected routerOptions: RouterOptions = {
-    defaultPage: "dashboard",
+    defaultPage: "picker",
     showLoading: true,
+    // Make sure that we have a config entry in the URL before rendering other pages
+    beforeRender: (page) => {
+      const searchParams = new URLSearchParams(window.location.search);
+      if (searchParams.has("config_entry")) {
+        this._configEntry = searchParams.get("config_entry");
+        searchParams.delete("config_entry");
+        let url = `${this.routeTail.prefix}${this.routeTail.path}`;
+        const search = searchParams.toString();
+        if (search) {
+          url += `?${search}`;
+        }
+        navigate(url, { replace: true });
+      }
+
+      if ((!page || page === "picker") && this._configEntry) {
+        return "dashboard";
+      }
```

Comment:
When trying to navigate back from the dashboard to the picker, `this._configEntry` will still be set, so instead of ending up back in the picker, you end up in the dashboard, again.

So, you click back in the dashboard, but nothing happens visually, until you click back a second time and you then skip backwards to the HA settings, instead of the picker.